core_ui/src/main/java/com/alexafit/core_ui/components/IconActionButton.kt

```diff
@@ -0,0 +1,28 @@
+package com.alexafit.core_ui.components
+
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Button
+import androidx.compose.material.Icon
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+
+@Composable
+fun IconActionButton(
+    onClick: () -> Unit,
+    modifier: Modifier = Modifier,
+    isEnabled: Boolean = true,
+    imageVector: ImageVector,
+    contentDescription: Int
```

Comment:
modifier should be the first optional parameter, but also, the required parameters (imageVector and contentDescription) should be first than the optional parameters

If not, to set an imageVector and contentDescription you have to use named parameters all the time

So I would:

```
imageVector: ImageVector,
contentDescription: Int,
onClick: () -> Unit,
modifier: Modifier = Modifier,
isEnabled: Boolean = true
```
In fact, if you have a lambda, you can set it as last parameter to make it so the Composable is a high-order function

FInal result:

```
imageVector: ImageVector,
contentDescription: Int,
modifier: Modifier = Modifier,
isEnabled: Boolean = true,
onClick: () -> Unit
```


Now you can call:

```
IconActionButton(vector, description) {
// Your onClick logic
}

```